test/e2e/plain-v0_provisioner_test.go

```diff
@@ -423,8 +423,118 @@ var _ = Describe("plain-v0 provisioner bundleinstance", func() {
 			}).Should(BeTrue())
 		})
 	})
+
+	When("a BundleInstance targets a valid bundle but the bundle contains resources that already exist", func() {
+		var (
+			bundle      *rukpakv1alpha1.Bundle
+			biOriginal  *rukpakv1alpha1.BundleInstance
+			biDuplicate *rukpakv1alpha1.BundleInstance
+			ctx         context.Context
+		)
+		BeforeEach(func() {
+			ctx = context.Background()
+
+			By("creating the testing Bundle resource")
+			bundle = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-crds-valid",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Image:                "quay.io/tflannag/olm-plain-bundle:olm-crds-v0.20.0",
+				},
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+
+			biOriginal = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-apis-original",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+
+			err = c.Create(ctx, biOriginal)
+			Expect(err).To(BeNil(), "failed to create original bundle instance")
+
+			biDuplicate = &rukpakv1alpha1.BundleInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "olm-apis-duplicate",
+				},
+				Spec: rukpakv1alpha1.BundleInstanceSpec{
+					ProvisionerClassName: plainProvisionerID,
+					BundleName:           bundle.GetName(),
+				},
+			}
+
+			err = c.Create(ctx, biDuplicate)
```

Comment:
I'm not sure but I think this may be a bit racy. Creating one object after the other sequentially like this doesn't necessarily mean that they will be created on the API server in the same order. I guess for a small kind cluster with no load we can expect it to be the case, but for example on Openshift it may not be always true that the second bundle is always seen after the first bundle if you create them at the same time. Maybe we could add an Eventually() or even sleep between creating the first and second BI, just to ensure the test is not prone to these types of timing issues?

Reply:
What if we had an Eventually stop gap between them, like you said, that attempts to grab each BI. It would guarantee that it is created and thus prevent proceeding before both are existing and ready.

Reply:
I think this is fine for now imo but Dan's point seems valid - we may need a couple of different helper functions to drive Bundle/BI resources creation at this point, and revisit refactoring the e2e suite structure I had built up.

Reply:
I'll make a todo comment for it 👍🏻

Reply:
https://github.com/operator-framework/rukpak/issues/138